src/context/json_schema_provider.js

```diff
@@ -47,9 +47,10 @@ class JSONSchemaFormDataProvider extends ContextProvider {
     get uiSchemaStyled() {
         const schema = this.uiSchema;
         return _.each(schema, (v, k, l) => {
-            l[k] = { ...v, ...this.defaultFieldStyles };
-            // retain any prior class attributes
-            l[k].classNames = `${v.classNames} ${this.defaultClassNames}`;
+            l[k].classNames = v.classNames
+                ? `${v.classNames} ${this.defaultClassNames}`
+                : this.defaultClassNames;
+            return null;
```

Comment:
Not sure what happened here...

Reply:
Previously, it would add `undefined` to the default class name if the UI schema did not have a `classNames` property for the given field. I just changed the logic to only add `classNames` to the string if it is actually defined.

While working on this, I found the behavior of this function a bit counterintuitive, and perhaps we want to consider changing it so that whenever `classNames` is present in the UI schema, it _overwrites_ rather than prepends the default class names.

Reply:
Regarding 2, `defaultFieldStyles` already contains `classNames` and it seemed redundant, but I will change it back so that one may implement a custom version locally (which is not currently the case).